### ISSUE TITLE
Revise methodology for crude oil production/export/import balancing

### ISSUE THREAD

**Issue by mkmahajan** [EnergyInnovation/eps-us#270]
A user pointed out the while crude oil production + imports - exports equals consumption in the BAU case, that does not hold for the NDC scenario. I believe our crude oil exports are currently too large because we're assigning the reduction of crude oil consumption to exports, but then double counting by also assigning the change in secondary petroleum product consumption multiplied by the "Percentage of Fuel Demand Change that Alters Exports" to increased exports. 

Additionally, 1 barrel of crude oil does results in less than 1 barrel of secondary petroleum products. So I don't think we want to keep using our variable "Change in Crude Exports due to Changes in Secondary Petroleum Product Use," which assumes a 1:1 relationship. Instead, I think we may want to isolate the change in crude consumption in the refining industry specifically, which is based on the change in production of secondary petroleum products, then apportion that between production/exports/imports.

**Comment by robbieorvis** [comment]
@mkmahajan is this still an outstanding issue?

**Comment by mkmahajan** [comment]
Yes this is still outstanding. I think we had previously discussed getting this into 4.0 only if we had time

**Comment by robbieorvis** [comment]
K. I will advance it to 4.1

**Comment by jrissman** [comment]
I would like to look into this carefully.  The code for handling this is somewhat intricate.

Note that refineries can have [refinery gain](https://www.eia.gov/energyexplained/oil-and-petroleum-products/refining-crude-oil-inputs-and-outputs.php) (also known as processing gain), where their outputs have higher volume than the crude oil inputs.  While the definition is based on volume, it _might_ also apply to energy content, since it is possible that some of the energy in the non-feedstock energy inputs to the refinery could end up contributing energy to the output products.  I feel like I've seen something along these lines in energy balance tables for some region, maybe the EU.  I've been wanting to get a really detailed understanding of refinery operations, which could be helpful here in making sure the code is handling everything correctly.
